dpkg improvements

dannf opened the pull request:
Make gnutar a runtime dependency + some minor test cleanups.

dannf committed:
dpkg: Make gnutar an explicit runtime dependency

Core functionality of dpkg is broken if gnutar isn't installed.
The current tests only pass because we explicitly install
in the test pipeline. Let's just make it a runtime dependency.

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

dannf committed:
dpkg: Future-proof test for additional architectures

If we use 'Architecture: all', we needn't try to calculate
the dpkg-architecture for the running host.

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

dannf committed:
dpkg: Split version/help tests into their own step

They are currently in the "Create a sample Debian package"
step, but that's not what they are testing. Really just
an excuse to coin a name for the version/help test pattern.

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

dannf committed:
dpkg: Bump epoch

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

justinvreeland commented on dpkg.yaml:
```diff
@@ -119,7 +122,6 @@ test:
         - bzip2
         - libmd
```
While we're here a lot of these look wrong for tests. I don't see any of these used directly so I'd expect the package to bring them in as part of the install.

dannf replied:
good call, fixed!

dannf committed:
dpkg: dpkg-dev: Add missing runtime dependencies

These are actual dependencies, they shouldn't need to be
specified by the test environment. Shared libraries
needn't be specified, they'll be added by melange.

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

dannf committed:
dpkg: make xz a runtime dependency instead of compensating w/ a test dep

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

dannf committed:
dpkg; Test all built-in compression types

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

dannf committed:
dpkg: enable zstd compression support

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>
